fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2093,29 +2095,97 @@ public void boundaryPrimaryKeys() throws Exception {
+                "the test is meaningless if the records are not across boundaries");
+        assertThat( boundaryPrimaryKeys.get(0), greaterThanOrEqualTo(Tuple.from(-25L * 39)));
+        assertThat( boundaryPrimaryKeys.get(boundaryPrimaryKeysSize - 1), lessThanOrEqualTo(Tuple.from(24L * 39)));
+        assertEquals(boundaryPrimaryKeys.stream().sorted().distinct().collect(Collectors.toList()), boundaryPrimaryKeys,
+                "the list should be sorted without duplication.");
+        for (Tuple boundaryPrimaryKey : boundaryPrimaryKeys) {
+            assertEquals(1, boundaryPrimaryKey.size(), "primary keys should be a single value");
+        }
+
+        // Test splitIndexBuildRange.
+        List<Pair<Tuple, Tuple>> oneRangePerSplit = getOneRangePerSplit(range, boundaryPrimaryKeys);
+
+        assertEquals(1, indexer.splitIndexBuildRange(Integer.MAX_VALUE, Integer.MAX_VALUE).size(),
+                "the range is not split when it cannot be split to at least minSplit ranges");
+        checkSplitIndexBuildRange(1, 2, null, indexer); // to test splitting into fewer than the default number of split points
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize - 2, null, indexer); // to test splitting into fewer than the default number of split points
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize, null, indexer);
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize + 1, oneRangePerSplit, indexer); // to test exactly one range for each split
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize + 2, oneRangePerSplit, indexer);
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, Integer.MAX_VALUE, oneRangePerSplit, indexer); // to test that integer overflow isn't a problem
+
+        indexer.close();
+    }
+
+    private List<Pair<Tuple, Tuple>> getOneRangePerSplit(TupleRange tupleRange, List<Tuple> boundaries) {
+        List<Tuple> newBoundaries = new ArrayList<>(boundaries);
+        if (tupleRange.getLow().compareTo(boundaries.get(0)) < 0) {
+            newBoundaries.add(0, tupleRange.getLow());
+        }
+        if (tupleRange.getHigh().compareTo(boundaries.get(boundaries.size() - 1)) > 0) {
+            newBoundaries.add(tupleRange.getHigh());
+        }
+
+        List<Pair<Tuple, Tuple>> oneRangePerSplit = new ArrayList<>();
+        for (int i = 0; i < newBoundaries.size() - 1; i++) {
+            oneRangePerSplit.add(Pair.of(newBoundaries.get(i), newBoundaries.get(i + 1)));
+        }
+        return oneRangePerSplit;
+    }
+
+    private void checkSplitIndexBuildRange(int minSplit, int maxSplit,
+                                           @Nullable List<Pair<Tuple, Tuple>> expectSplitRanges,
+                                           OnlineIndexer indexer) throws Exception {
+        logger.info("wawawa " + minSplit + " " + maxSplit + " " + expectSplitRanges);
```

Comment:
Logging left from debugging?

I'm not opposed to their being logging here, but maybe not "wawawa".

Reply:
Oh, I forgot to remove it